doc/user-manual/getting-started/hello-world.lagda.rst

```diff
@@ -7,35 +7,417 @@
+were able to state the property in a way that Agda understands, our objective
+is to prove it.
+
+First, we will need to import the constructors ``zero`` and ``suc`` of the
+already imported datatype ``Nat`` and the definition ``cong`` from the
+`standard library <std-lib_>`_..
+
+.. code-block:: agda
+
+  open import Agda.Builtin.Nat using (zero ; suc)
+  open import Relation.Binary.PropositionalEquality using (cong)
+  open import Agda.Builtin.Equality using (refl)
+
+In order to prove ``+assoc-enun`` we need just to find an element of that type.
+We will name this element ``+assoc``, but like always one can go ahead and try
+to find a better name, or a name that is more suitable for a certain context.
+
+.. code-block:: agda
+
+  +assoc : ∀ (x y z : Nat) → x + (y + z) ≡ (x + y) + z
+
+If we load now the file, Agda will complain. The name ``+assoc`` was declared
+correctly but a definition was not provided. That definition is actually the
+proof that we are looking for. To build the definition, we need to know more
+about holes and case splitting.
+
+Holes and case splitting
+------------------------
+
+Agda will help us to find the proof by using its interactive mode. We will
+first write a very simple clause so the file can be loaded even if we still
+do not know the proof. The clause consists of the name of the property, the
+input variables, the symbol equal ``=`` and the question mark ``?``.
+
+.. code-block:: agda
+
+  +assoc x y z = ?
+
+Now Agda is not throwing an error when loading the file, but returning
+***All Goals***. We have entered the interactive proving mode. Agda turns
+our question mark into what is called a *hole* ``{!  0!}``. The number
+``0`` inside labels the goal.
+
+The next step would be choosing the pattern variable and perform case
+splitting on it. Put the cursor inside the hole and press ``C-c C-c``.
+Agda will ask for the pattern variable, let's write ``x`` and press
+``Return``.
+
+.. code-block:: agda
+
+  +assoc zero y z = {!  0!}
+  +assoc (suc x) y z = {!  1!}
+
+Agda performs the case splitting of the clause, now we have one clause for
+the case ``zero`` and another for the case ``suc x``. That means also that
+we have two holes. The first one is easy to resolve, because when the case
+of ``x`` is ``zero``, the equivalence that we want to prove holds
+definitionally.
+
+.. note:: The case splitting on the variable ``x`` is complete.
+  Proving the definition for ``zero`` and ``suc x`` amounts to proving it
+  for every ``x : Nat``.
+
+Put the cursor inside the first hole labeled ``0`` and press ``C-c C-r``
+to resolve it.
 
 .. code-block:: agda
 
-  module hello-world where
+  +assoc x y z = refl
+  +assoc (suc x) y z = {!  1!}
+
+Now we have again one hole to resolve. If you load the file again, you will
+get the type of the term that should be in the hole
+``?0 : suc x + (y + z) ≡ suc x + y + z``. It seems like proving ``+assoc``
```

Comment:
This is the first time that you rely essentially on type-level evaluation of a function, which is something that can be very non-obvious to newcomers. However, you ignore it here completely. It might make sense to introduce the normalize command `C-c C-n` or even the (ugly) prefix `C-u C-u` so you can see the computation happening: `suc x + (y + z)` computes to `suc (x + (y + z))` and `suc x + y + z` computes to `suc (x + y + z)`.

Reply:
It might also make sense to note that Agda prints the term `(x + y) + z` as `x + y + z` thanks to the infix statement in the builtin `Nat` module.